src/AccessibilityInsights.CommonUxComponents/Controls/FabricIconControl.xaml.cs

```diff
@@ -0,0 +1,3278 @@
+    
+        public static readonly DependencyProperty GlyphNameProperty =
+            DependencyProperty.Register("GlyphName", typeof(FabricIcon?), typeof(FabricIconControl), new PropertyMetadata(default(FabricIcon?), OnGlyphNameChanged));
+
+        public FabricIcon? GlyphName
+        {
+            get { return GetValue(GlyphNameProperty) as FabricIcon?; }
+
+            set { SetValue(GlyphNameProperty, value); }
+        }
+
+        public static void OnGlyphNameChanged(DependencyObject o, DependencyPropertyChangedEventArgs e)
+        {
+            FabricIconControl sender = o as FabricIconControl;
+
+            if (sender != null && sender.GlyphName.HasValue)
+            {
+                sender.lblIcon.Text = FabricIconControlValues.GetIcon(sender.GlyphName.Value);
+            }
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Size of icon 
+        /// </summary>
+        #region GlyphSize (Dependency Property)
+
+        public static readonly DependencyProperty GlyphSizeProperty =
+            DependencyProperty.Register("GlyphSize", typeof(GlyphContext), typeof(FabricIconControl), new PropertyMetadata(GlyphContext.Default, OnGlyphSizeChanged));
+
+        public GlyphContext GlyphSize
+        {
+            get { return (GlyphContext)GetValue(GlyphSizeProperty); }
+
+            set { SetValue(GlyphSizeProperty, value); }
+        }
+
+        public static void OnGlyphSizeChanged(DependencyObject o, DependencyPropertyChangedEventArgs e)
+        {
+            FabricIconControl sender = o as FabricIconControl;
+
+            if (sender != null)
+            {
+                if (sender.GlyphSize == GlyphContext.Custom)
+                {
+                    sender.lblIcon.FontSize = sender.FontSize;
+                }
+                else
+                {
+                    sender.lblIcon.FontSize = GlyphSizeValues.GetGlyphSize(sender.GlyphSize);
+                }
+            }
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Size of icon if custom GlyphSize chosen
+        /// </summary>
+        #region FontSize (Dependency Property)
+
+        new public static readonly DependencyProperty FontSizeProperty =
+            DependencyProperty.Register("FontSize", typeof(double), typeof(FabricIconControl), new PropertyMetadata(1.0, OnFontSizeChanged));
+
+        new public double FontSize
+        {
+            get { return (double)GetValue(FontSizeProperty); }
+
+            set { SetValue(FontSizeProperty, value); }
+        }
+
+        public static void OnFontSizeChanged(DependencyObject o, DependencyPropertyChangedEventArgs e)
+        {
+            FabricIconControl sender = o as FabricIconControl;
+
+            if (sender != null)
+            {
+                if (sender.GlyphSize != GlyphContext.Custom)
+                {
+                    //throw new NotSupportedException("GlyphSize must be set to 'Custom' in order to directly set FontSize.");
```

Comment:
This looks incomplete

Reply:
it was just copied from the DesktopUI so I believe it should be fine
@jalkire, any thought?